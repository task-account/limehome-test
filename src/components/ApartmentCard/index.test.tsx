import renderer from "react-test-renderer";
import { Apartment } from "@models";

import { ApartmentCard } from ".";

const mockApartment: Apartment = {
  id: 96,
  external_id: "AQ2",
  review_widget_id: "elfsight-app-1a175e41-0a9b-42ad-8db6-b660a0a178f9",
  price: "55â‚¬",
  rating: "4.5",
  name: "montabaur am quendelberg",
  city: "montabaur",
  city_id: 30,
  street: "am quendelberg",
  location: {
    lat: 50.4336675,
    lng: 7.830375699999999,
    city: "Montabaur",
    postalCode: "56410",
    countryCode: "DE",
    addressLine1: "Am Quendelberg 2",
    countryName: "Germany"
  },
  distance: 0.4043757328515891,
  images: [
    {
      url: "https://limehome.imgix.net/properties/96/3762f856-fce1-454c-b71b-b4f00b06e1c4.jpg",
      is_portrait: false,
      position: 0,
      unit_group_ids: [13]
    }
  ],
  lowest_price_per_night: null,
  lowest_price_per_month: null,
  unit_groups: [
    {
      id: 13,
      bedroom_count: 1,
      lowest_price_per_night: null,
      lowest_price_per_month: null
    }
  ]
};

jest.mock("@utils/hooks", () => ({
  withStyles: () => () => ({ useStyles: jest.fn() })
}));

jest.mock("@react-navigation/native", () => {
  const actualModule = jest.requireActual("@react-navigation/native");
  return {
    ...actualModule,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn()
    })
  };
});

describe(`ApartmentCard`, () => {
  it("render", () => {
    const tree = renderer.create(<ApartmentCard apartment={mockApartment} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
