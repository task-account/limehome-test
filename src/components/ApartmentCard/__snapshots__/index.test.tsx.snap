// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApartmentCard render 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <View>
      <Image
        onError={[Function]}
        onLoad={[Function]}
        onLoadEnd={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        source={
          Object {
            "uri": "https://limehome.imgix.net/properties/96/3762f856-fce1-454c-b71b-b4f00b06e1c4.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View>
      <Text>
        4.5
      </Text>
      <View
        style={
          Object {
            "flex": undefined,
            "marginBottom": undefined,
            "marginHorizontal": 2,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "marginVertical": undefined,
          }
        }
      />
      <Text />
    </View>
  </View>
  <View>
    <Text
      numberOfLines={1}
    >
      montabaur am quendelberg
    </Text>
    <View
      style={
        Object {
          "flex": undefined,
          "marginBottom": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "marginVertical": 4,
        }
      }
    />
    <View>
      <Text />
      <View
        style={
          Object {
            "flex": undefined,
            "marginBottom": undefined,
            "marginHorizontal": 4,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "marginVertical": undefined,
          }
        }
      />
      <Text>
        0.4 km from city centre
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#4b4b4d",
          "height": 1,
          "marginBottom": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "marginVertical": 8,
        }
      }
    />
    <View>
      <Text>
        From
      </Text>
      <View
        style={
          Object {
            "flex": undefined,
            "marginBottom": undefined,
            "marginHorizontal": 4,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "marginVertical": undefined,
          }
        }
      />
      <Text>
        55â‚¬/Night
      </Text>
    </View>
  </View>
</View>
`;
